CouplingDataUser.C

```diff
@@ -31,11 +31,32 @@ void preciceAdapter::CouplingDataUser::setPatchIDs(std::vector<int> patchIDs)
     patchIDs_ = patchIDs;
 }
 
-void preciceAdapter::CouplingDataUser::setLocationsType(std::string locationsType)
+void preciceAdapter::CouplingDataUser::setLocationsType(LocationType locationsType)
 {
     locationsType_ = locationsType;
 }
 
+void preciceAdapter::CouplingDataUser::checkDataLocation(const bool meshConnectivity)
+{
+    if (this->isLocationTypeSupported(meshConnectivity) == false)
+    {
+        std::string location("none");
+        if (locationsType_ == LocationType::faceCenters)
+            location = "faceCenters";
+        else if (locationsType_ == LocationType::faceNodes)
+            location = "faceNodes";
+
+        FatalErrorInFunction
+            << "ERROR: the configured location type "
+            << location
+            << " is not supported for the data "
+            << this->getDataName()
+            << ". Please select a different location type or "
+            << "a different data set."
+            << exit(Foam::FatalError);
```

Comment:
Why not using our own error handling utility?

Reply:
Good hint.

Reply:
I copied this actually from another function in the adapter.